Hearthstone/app/src/main/java/com/example/hearthstone/ui/homepage/HomeFragment.kt

```diff
@@ -1,40 +1,65 @@
-package com.example.hearthstone.ui.HomePage
+package com.example.hearthstone.ui.homepage
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.navigation.findNavController
 import com.example.hearthstone.MainActivity
 import com.example.hearthstone.R
 import com.example.hearthstone.adapter.HearthstoneAdapter
 import com.example.hearthstone.data.network.repo.HSRepo
 import com.example.hearthstone.databinding.FragmentHomeBinding
 
+
 class HomeFragment : Fragment() {
 
     private lateinit var binding: FragmentHomeBinding
     private lateinit var viewModel: HomeViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-    }
-
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         binding = DataBindingUtil.inflate(inflater, R.layout.fragment_home, container, false)
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        viewModel = HomeViewModel(MainActivity.ourApplication, HSRepo.provideHSRepoApi(), view)
+        viewModel = HomeViewModel(MainActivity.ourApplication, HSRepo.provideHSRepoApi())
```

Comment:
Are you going to be using DI soon?

Reply:
Yes, the DI is in the next PR in which I only included the DP